### PR TITLE
Use settings.DEPLOY_ENVIRONMENT in error email subject

### PR DESCRIPTION
**Pull request by chosak**
Instead of using the environment variable `EMAIL_SUBJECT_PREFIX`, which we don't seem to reliably set, use the `DEPLOY_ENVIRONMENT` setting (which comes from the `DEPLOY_ENVIRONMENT` variable) instead. If this setting isn't set, fallback to the default (`"[Django] "`).

## Changes

- Deprecate use of `EMAIL_SUBJECT_PREFIX` for error email subjects in favor of `DEPLOY_ENVIRONMENT`.

## Checklist

* [X] PR has an informative and human-readable title
* [X] Changes are limited to a single goal (no scope creep)
* [X] Code can be automatically merged (no conflicts)
* [X] Passes all existing automated tests
* [ ] Project documentation has been updated
* [X] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by higs4281** [comment]
Great idea.

One very minor thought: A lowercase subject prefix might look odd. `.title()`?

File: cfgov/cfgov/settings/base.py [hunk 1]
```diff
@@ -376,7 +376,10 @@
 
 # Email
 ADMINS = admin_emails(os.environ.get('ADMIN_EMAILS'))
-EMAIL_SUBJECT_PREFIX = os.environ.get('EMAIL_SUBJECT_PREFIX')
+
+if DEPLOY_ENVIRONMENT:
+    EMAIL_SUBJECT_PREFIX = u'[{}] '.format(DEPLOY_ENVIRONMENT)
+
```
Comment (richaagarwal) [inline cfgov/cfgov/settings/base.py]: is there supposed to be an `else` here to set it to a default, or is that happening somewhere else?

Comment (willbarton) [inline cfgov/cfgov/settings/base.py]: [The default is just `[Django]`](https://docs.djangoproject.com/en/1.8/ref/settings/#email-subject-prefix), which is less than ideal but no worse than we have now. I think that's acceptable.

**Comment by chosak** [comment]
@higs4281 thanks, that's a great suggestion. Addressed in 4ce0728.

**Pull request merged**